net/tstun/wrap.go

```diff
@@ -481,6 +494,16 @@ func (t *Wrapper) filterIn(buf []byte) filter.Response {
 				f(data)
 			}
 		}
+	case ipproto.UDP:
+		if pkt := p.Payload(); disco.LooksLikeDiscoWrapper(pkt) {
```

Comment:
Some inversion of control to reduce indentation would be nice here.

```go
pkt := p.Payload()
if !disco.LooksLikeDiscoWrapper(pkt) {
  break
}
// etc
```

Reply:
pulled out